src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32.cs

```diff
@@ -115,12 +116,27 @@ public string SetSubtitle(uint AppType, IPublicAPI api)
             {
                 return api.GetTranslation("powertoys_run_plugin_program_web_application");
             }
+            else if(AppType == (uint)ApplicationTypes.RUN_COMMAND)
+            {
+                return api.GetTranslation("powertoys_run_plugin_program_run_command");
+            }
             else
             {
                 return String.Empty;
             }
         }
 
+        public bool FilterPartialMatchForRunCommands(string query)

```

Comment:
Is this function required to be `public`?

Reply:
This functionality is being unit tested. I could create another public function which calls this private function but that seems like an unnecessary overhead. Let me know if you have any thoughts.

Reply:
My concern was that we are making a function public for the sole purpose of testing it. It would be better if we could test this functionality through public methods. One such example would be to test if `Result(string query, IPublicAPI api)` function returns a Run program in case there is a full match or doesn't return any program if there is partial match.

Reply:
I'm not sure if unit testing Result is the right way to go about because it is not a single piece of code that we would be testing, rather it calls a lot of functions and has a lot of logic going on even within itself. 

I understand that we don't want to make a function public for the sole reason of testing, but it seems like if we do want to unit test the parent `Result` function instead of the individual components, then this must be done for all the unit tests in the ProgramPluginTest.cs file (Which is testing individual components right now and not the main `Result` function). This requires a lot of refactoring and can maybe be done as a separate PR?

@ryanbodrug-microsoft, do you have any suggestions on the best way of unit testing all the functionalities which are presently being tested in the `ProgramPluginTest.cs` file?

Reply:
@alekhyareddy28 / @somil55.  I think the way it is is fine.  I wouldn't spend the effort to refactor the tests.  Its not ideal to make methods public for the sole purpose of testing, but IMHO the contract that is being enforced here doesn't leak any abstraction that should be kept as as an implementation detail for the Object under test.